### PR TITLE
Better `CE` error reporting when using `use!` with `and!`

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

- Improves the error range for `use! may not be combined with and!`
- Updates `SynExprAndBangTrivia` to include also `AndBangKeyword` range
- Uses only the `and!` range since it's what adds the error in the first place.

### Before

Screenshot 2024-09-08 at 16 32 46

### After

```fsharp
let run r2 r3 =
    result {
        use! b = r2
        and! c = r3
        ^^^^
        return b - c
    }
```

## Checklist

- [x] Test cases added
- [x] Release notes entry updated

### CODE REVIEW

**Comment by auduchinok** [comment]
@edgarfgp Could it be better to highlight the `and!` part, since it's what adds the error? Could it also highlight the keyword only?

**Comment by edgarfgp** [comment]
> Could it be better to highlight the `and!` part, since it's what adds the error? Could it also highlight the keyword only?

Yeah I think that makes more sense. Thanks for the review.

**Comment by edgarfgp** [comment]
@auduchinok I played around with your suggestion and I was able to update `SynExprAndBangTrivia` to also include `and!` range.

File: src/Compiler/SyntaxTree/SyntaxTree.fsi [hunk 1]
```diff
@@ -982,6 +982,12 @@ type SynExprAndBang =
         range: range *
         trivia: SynExprAndBangTrivia
 
+    /// Gets the syntax range of this construct
+    member Range: range
+
+    /// Gets the trivia associated with this construct
+    member Trivia: SynExprAndBangTrivia
```
Comment (edgarfgp) [inline src/Compiler/SyntaxTree/SyntaxTree.fsi:989]: Should we include the trivia. Or just take the start range + 4 as `and!` length wont change. So we can avoid having this addition ?

Comment (baronfel) [reply src/Compiler/SyntaxTree/SyntaxTree.fsi:989]: IMO we should take it. Correctness is good.

**Review by psfinaki — APPROVED** [review]
Good stuff, I like this approach with adding trivia. Thanks for making diags better every week :)

**Review by vzarytovskii — APPROVED** [review]

**Comment by edgarfgp** [comment]
@psfinaki I have more planned for CE in the upcoming weeks

**Pull request merged**